{"Functions": {"toString()": {"Returns": [["@B_java#lang#String_E@", "a string representation of this throwable."]], "Parameters": [], "Throws": [], "SeeAlso": [], "Permissions": [], "Description": "Returns a short description of this throwable. The result is the concatenation of: the \": \" (a colon and a space) the result of invoking this object's @B_java#lang#Throwable#getLocalizedMessage_E@ method If getLocalizedMessage returns null, then just the class name is returned.", "history": "Added in API level 1", "FullName": "public String toString ()"}, "getException()": {"Returns": [["@B_java#lang#Exception_E@", "the exception thrown by the privileged computation that resulted in this PrivilegedActionException."]], "Parameters": [], "Throws": [], "SeeAlso": ["@B_java#security#PrivilegedExceptionAction_E@", "@B_java#security#AccessController#doPrivileged_E@"], "Permissions": [], "Description": "Returns the exception thrown by the privileged computation that resulted in this PrivilegedActionException. This method predates the general-purpose exception chaining facility. The @B_java#lang#Throwable#getCause_E@ method is now the preferred means of obtaining this information.", "history": "Added in API level 1", "FullName": "public Exception getException ()"}, "PrivilegedActionException(java.lang.Exception)": {"Returns": [], "Parameters": [["@B_java#lang#Exception_E@", "Exception: The exception thrown"]], "Throws": [], "SeeAlso": [], "Permissions": [], "Description": "Constructs a new PrivilegedActionException \"wrapping\" the specific Exception.", "history": "Added in API level 1", "FullName": "public PrivilegedActionException (Exception exception)"}, "getCause()": {"Returns": [["@B_java#lang#Throwable_E@", "the cause of this exception."]], "Parameters": [], "Throws": [], "SeeAlso": [], "Permissions": [], "Description": "Returns the cause of this exception (the exception thrown by the privileged computation that resulted in this PrivilegedActionException).", "history": "Added in API level 1", "FullName": "public Throwable getCause ()"}}, "Inheritance": ["@B_java#lang#Throwable_E@", "@B_java#lang#Exception_E@"], "ClassName": "java.security.PrivilegedActionException", "ClassDesc": "Legacy security code; do not use. This exception is thrown by doPrivileged(PrivilegedExceptionAction) and doPrivileged(PrivilegedExceptionAction, AccessControlContext context) to indicate that the action being performed threw a checked exception. The exception thrown by the action can be obtained by calling the getException method. In effect, an PrivilegedActionException is a \"wrapper\" for an exception thrown by a privileged action. As of release 1.4, this exception has been retrofitted to conform to the general purpose exception-chaining mechanism. The \"exception thrown by the privileged computation\" that is provided at construction time and accessed via the @B_java#security#PrivilegedActionException#getException_E@ method is now known as the @B_java#lang#Throwable#getCause_E@ method, as well as the aforementioned \"legacy method.\""}